akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,22 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case sel @ ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.head match {
+        case SelectChildName(name) ⇒
+          val child = getChild(name)
+          if (elements.tail.isEmpty) {
+            child ! msg
+          } else {
+            child ! sel.copy(elements = elements.tail)
+          }
+        case _ ⇒ // .. and * are unsupported
```

Comment:
To get a reference on `EventStream` can I use `provider.systemGuardian.underlying.system.eventStream` or is there a better way?
The other way I found was matching on getParent, but I think it's ugly
```scala
getParent match {
  case refWithCell: ActorRefWithCell ⇒ {
    val emptyRef = new EmptyLocalActorRef(provider, path / sel.elements.map(_.toString), refWithCell.underlying.system.eventStream)
    emptyRef.tell(msg, sender)
  }
  case _ ⇒
    this.provider.deadLetters.tell(msg, sender)
}
```